src/irmin-pack/layered/ext_layered.ml

```diff
@@ -755,34 +759,24 @@ struct
     end
   end
 
-  let dump_stats msg =
-    let stats = Irmin_layers.Stats.get () in
-    Log.debug (fun l ->
-        l "%s: contents=%d, nodes=%d, commits=%d, skips=%d" msg
-          (List.hd stats.copied_contents)
-          (List.hd stats.copied_nodes)
-          (List.hd stats.copied_commits)
-          stats.skips)
-
-  let with_stats msg f = f >|= fun () -> dump_stats msg
-
   let copy ~cancel ~min ~max ~squash ~upper:(copy_in_upper, min_upper) t =
     (* Copy commits to lower: if squash then copy only the max commits.
        In case cancellation of the freeze, copies to the lower layer will not
        be reverted. Since the copying is performed in the [rev] order, the next
        freeze will resume copying where the previous freeze stopped. *)
+    Irmin_layers.Stats.freeze_section "copy to lower";
     (if t.X.Repo.with_lower then
      let min = if squash then max else min in
-     with_stats "copied in lower" (Copy.CopyToLower.copy ~cancel t ~min max)
```

Comment:
this print allowed us to estimate the number of objects copied to lower vs the ones copied to upper. We don't necessarily need to have this info all the time; but it would be nice to have a way to only count the objects copied in lower for instance, if we need that at some point. (We can also do this in a separate PR, as its not immediately useful).

Reply:
see latest version